Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -173,7 +176,7 @@ class TopTabsViewController: UIViewController {
     }
     
     func privateTabTapped() {
-        delegate?.topTabsDidPressPrivateTab()
+        delegate?.topTabsDidPressPrivateTab(isPrivate ? lastNormalTab : lastPrivateTab)
```

Comment:
This... seems weird that we are passing in a normal tab as the parameter to `topTabsDidPressPrivateTab`. Either the method needs to be renamed or what you pass to it needs to changed.